src/Tutorials/GausianProcessRegressor.cs

```diff
@@ -39,35 +40,51 @@ public void Run()
-            prior.ObservedValue = new SparseGP(new SparseGPFixed(gp, trainingInputs.ToArray()));
-            double logOdds = engine.Infer<Bernoulli>(evidence).LogOdds;
-            Console.WriteLine("{0} evidence = {1}", kf, logOdds.ToString("g4"));
-
-            // Infer the posterior Sparse GP
-            SparseGP sgp = engine.Infer<SparseGP>(f);
-            //PlotPredictions(sgp, trainingInputs, trainingOutputs);
+            for(var i = 0; i < 2; i++)
+            {
+                // Modelling code
+                Variable<bool> evidence = Variable.Bernoulli(0.5).Named("evidence");
+                IfBlock block = Variable.If(evidence);
+                Variable<SparseGP> prior = Variable.New<SparseGP>().Named("prior");
+                Variable<IFunction> f = Variable<IFunction>.Random(prior).Named("f");
+                VariableArray<Vector> x = Variable.Observed(trainingInputs).Named("x");
+                Range j = x.Range.Named("j");
+                VariableArray<double> y = Variable.Observed(trainingOutputs, j).Named("y");
+
+                var likelihoodDescription = string.Empty;
+                if (i == 0)
+                {
+                    // Standard Gaussian Process
+                    var score = GetScore(x, f, j);
+                    y[j] = Variable.GaussianFromMeanAndVariance(score, 0.8);
+                }
+                else
+                {
+                    // Gaussian Process with Student-t likelihood
+                    var noisyScore = GetNoisyScore(x, f, j, trainingOutputs);
+                    y[j] = Variable.GaussianFromMeanAndVariance(noisyScore[j], 0.8);
+                    likelihoodDescription = " with Student-t likelihood";
+                }
+                Console.WriteLine("Training a Gaussian Process regressor{0}", likelihoodDescription);
+                block.CloseBlock();
+
+                // Log length scale estimated as -1
+                var kf = new SquaredExponential(-1);
+                GaussianProcess gp = new GaussianProcess(new ConstantFunction(0), kf);
+
+                // Convert SparseGP to full Gaussian Process by evaluating at all the training points
+                prior.ObservedValue = new SparseGP(new SparseGPFixed(gp, trainingInputs.ToArray()));
+                double logOdds = engine.Infer<Bernoulli>(evidence).LogOdds;
+                Console.WriteLine("{0} evidence = {1}", kf, logOdds.ToString("g4"));
+
+                // Infer the posterior Sparse GP
+                SparseGP sgp = engine.Infer<SparseGP>(f);
+                //PlotPredictions(sgp, trainingInputs, trainingOutputs, likelihoodDescription);
+            }
         }
-#if oxyplot
-        private void PlotPredictions(SparseGP sgp, Vector[] trainingInputs, double[] trainingOutputs)
+
+        #if oxyplot
+        private void PlotPredictions(SparseGP sgp, Vector[] trainingInputs, double[] trainingOutputs, string likelihoodDescription)
```

Comment:
NIT: Reusing the `likelihoodDescription` variable will give some ugly file names; maybe have a conditional and just use `GPRegressionPredictionsStudentT.png` instead?